src/DynamoCore/Engine/CodeGeneration/AstCompilationEvents.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+
+namespace Dynamo.Engine.CodeGeneration
+{
+    /// <summary>
+    /// Provides AST compilation related events. Clients can provide more AST
+    /// to be included during compilation events.
+    /// </summary>
+    public static class AstCompilationEvents
+    {
+        /// <summary>
+        /// Compilation for a node has begun.
+        /// </summary>
+        public static event EventHandler<CompilationEventArgs> PreCompilation;
```

Comment:
are these public static events used currently? What would an example of using them look like? Should they really be static - what if there is more than one DynamoModel instantiated?